### PR TITLE
Delete Vector Search Endpoint when we get an error while waiting for creation

### PR DESCRIPTION
**Pull request by alexott**
## Changes

If we get an error while waiting for the endpoint creation we simply return that error.  But when we got a timeout, we didn't clean the endpoint and this led to the resource not being destroyed in the tests as the backend continued to provision it.

This PR deletes the vector search endpoint when we get a creation error during the wait phase. This behaviour is similar to the current behaviour of clusters.

## Tests

- [x] `make test` run locally
- [ ] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

File: vectorsearch/resource_vector_search_endpoint.go
Comment (tanmay-db) [inline vectorsearch/resource_vector_search_endpoint.go:51]: 10*time.Second for this, let's have another variable.

File: vectorsearch/resource_vector_search_endpoint.go
Comment (tanmay-db) [inline vectorsearch/resource_vector_search_endpoint.go:53]: Is this behaviour consistent? i.e. deleting works even if creation fails? This sounds strange and probably should be fixed on backend. Maybe it is similar to cancelling while creation is in progress. I think we should confirm this with the API owning team.

File: vectorsearch/resource_vector_search_endpoint.go
Comment (tanmay-db) [inline vectorsearch/resource_vector_search_endpoint.go:54]: instead of err2, we could probably use nestedErr

File: vectorsearch/resource_vector_search_endpoint.go
Comment (tanmay-db) [inline vectorsearch/resource_vector_search_endpoint.go:51]: instead of ctx2, we could probably use another variable, maybe: `ctxWithDeadline`

**Review by tanmay-db — APPROVED** [review]
Some comments but overall LGTM, deleting end point when it's not successfully created sounds strange and probably should be fixed at backend. If this behaviour is consistent and expected from backend team (and won't change in future) then this sounds good. cc: @mgyucht

Also, does the integration test pass now? 

Let's discuss this once in TF office hours before merging.

File: vectorsearch/resource_vector_search_endpoint.go
Comment (mgyucht) [inline vectorsearch/resource_vector_search_endpoint.go:50]: Technically, we should also subtract the time it took to make the create call. Otherwise, if it is longer than `deleteCallTimeout`, we will wait past the parent deadline.

**Review by mgyucht — APPROVED** [review]
One small nit, otherwise LGTM. I'll try to work on the abstraction tomorrow.

**Pull request merged**
